sdktests/server_side_eval.go

```diff
@@ -95,7 +95,7 @@ func RunParameterizedServerSideClientNotReadyEvalTests(t *ldtest.T) {
 			t.Run("evaluate flag without detail", func(t *ldtest.T) {
 				result := client.EvaluateFlag(t, servicedef.EvaluateFlagParams{
 					FlagKey:      flagKey,
-					User:         &user,
+					User:         user,
```

Comment:
I changed the structs in `servicedef` to take just a User rather than a pointer to a User. The only reason it was a pointer was that I was trying to think ahead to reusing some of this stuff for client-side SDKs, where the user is not a parameter. But I think that was premature— I just don't know enough yet about how these tests will be adapted for client-side SDKs— and it made the calling logic inconvenient.